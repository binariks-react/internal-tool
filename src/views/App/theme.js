export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
