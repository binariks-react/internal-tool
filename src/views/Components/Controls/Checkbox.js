import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const CheckboxWrapper = styled.div`
    position: relative;
    line-height: 1.5rem;
    min-height: 1.5rem;
`;

const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
    box-sizing: border-box;
    padding: 0;
    position: absolute;
    z-index: -1;
    opacity: 0;
`;

const Label = styled.label`
    display: flex;
    align-items: center;
    cursor: ${({ disabled }) => disabled ? 'not-allowed' : 'pointer'}

`;

const Text = styled.span`
    margin-left: 8px;
    font-size: 1rem;
    color: ${({ theme }) => theme.colors.text};
    opacity: ${({ disabled }) => disabled ? 0.7 : 1};
`;

const Icon = styled.svg`
    fill: none;
    stroke: transparent;
    transform: scale(0);
    stroke-width: 3px;
    transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
    transition-delay: .1s;
`;

const CustomCheckbox = styled.div`
    display: inline-block;
    width: 1.125rem;
    height: 1.125rem;
    border: 1px solid #e1e5eb;
    border-radius: 2px;
    transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
    box-shadow: none;
    background-color: white;
    text-align: center;
    
    input:checked + & {
      background-color: ${({ theme }) => theme.colors.secondary};
    
      svg {
        stroke: white;
        transform: scale(1);
      } 
    }
`;

const Checkbox = props => (
  <CheckboxWrapper {...props}>
    <Label {...props}>
      <HiddenCheckbox {...props} />
      <CustomCheckbox {...props}>
        <Icon viewBox="0 0 24 24">
          <polyline points="20 6 9 17 4 12" />
        </Icon>
      </CustomCheckbox>
      <Text {...props}>
        {props.label}
      </Text>
    </Label>
  </CheckboxWrapper>
);

Checkbox.defaultProps = {
  label: '',
  dafaultChecked: false,
  disable: false,
};

Checkbox.propTypes = {
  label: PropTypes.string,
  dafaultChecked: PropTypes.bool,
  disable: PropTypes.bool,
};


export default Checkbox;
