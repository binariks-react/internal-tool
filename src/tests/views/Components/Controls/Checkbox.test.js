import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from 'enzyme';
import theme from 'views/App/theme';
import Checkbox from 'views/Components/Controls/Checkbox';
import 'jest-styled-components';

describe('Checkbox', () => {
  it('common', () => {
    const checkbox = render(
      <ThemeProvider theme={theme}>
        <Checkbox />
      </ThemeProvider>
    );
    expect(checkbox).toMatchSnapshot();
  });

  it('with disabled', () => {
    const checkbox = render(
      <ThemeProvider theme={theme}>
        <Checkbox disable />
      </ThemeProvider>
    );
    expect(checkbox).toMatchSnapshot();
  });

  it('as checked', () => {
    const checkbox = render(
      <ThemeProvider theme={theme}>
        <Checkbox defaultChecked />
      </ThemeProvider>
    );
    expect(checkbox).toMatchSnapshot();
  });
});
