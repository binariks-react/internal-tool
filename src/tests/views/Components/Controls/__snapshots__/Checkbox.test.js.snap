// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox as checked 1`] = `
.c0 {
  position: relative;
  line-height: 1.5rem;
  min-height: 1.5rem;
}

.c2 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c5 {
  margin-left: 8px;
  font-size: 1rem;
  color: #5a6169;
  opacity: 1;
}

.c4 {
  fill: none;
  stroke: transparent;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  stroke-width: 3px;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  -webkit-transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  -webkit-transition-delay: .1s;
  transition-delay: .1s;
}

.c3 {
  display: inline-block;
  width: 1.125rem;
  height: 1.125rem;
  border: 1px solid #e1e5eb;
  border-radius: 2px;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  box-shadow: none;
  background-color: white;
  text-align: center;
}

input:checked + .c3 {
  background-color: #5a6169;
}

input:checked + .c3 svg {
  stroke: white;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div
  class="c0"
  label=""
>
  <label
    class="c1"
    label=""
  >
    <input
      checked=""
      class="c2"
      label=""
      type="checkbox"
    />
    <div
      class="c3"
      label=""
    >
      <svg
        class="c4"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
    <span
      class="c5"
      label=""
    />
  </label>
</div>
`;

exports[`Checkbox common 1`] = `
.c0 {
  position: relative;
  line-height: 1.5rem;
  min-height: 1.5rem;
}

.c2 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c5 {
  margin-left: 8px;
  font-size: 1rem;
  color: #5a6169;
  opacity: 1;
}

.c4 {
  fill: none;
  stroke: transparent;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  stroke-width: 3px;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  -webkit-transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  -webkit-transition-delay: .1s;
  transition-delay: .1s;
}

.c3 {
  display: inline-block;
  width: 1.125rem;
  height: 1.125rem;
  border: 1px solid #e1e5eb;
  border-radius: 2px;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  box-shadow: none;
  background-color: white;
  text-align: center;
}

input:checked + .c3 {
  background-color: #5a6169;
}

input:checked + .c3 svg {
  stroke: white;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div
  class="c0"
  label=""
>
  <label
    class="c1"
    label=""
  >
    <input
      class="c2"
      label=""
      type="checkbox"
    />
    <div
      class="c3"
      label=""
    >
      <svg
        class="c4"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
    <span
      class="c5"
      label=""
    />
  </label>
</div>
`;

exports[`Checkbox with disabled 1`] = `
.c0 {
  position: relative;
  line-height: 1.5rem;
  min-height: 1.5rem;
}

.c2 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.c5 {
  margin-left: 8px;
  font-size: 1rem;
  color: #5a6169;
  opacity: 0.7;
}

.c4 {
  fill: none;
  stroke: transparent;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  stroke-width: 3px;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  -webkit-transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: transform 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06),-webkit-transform 250ms cubic-bezier(.27,.01,.38,1.06);
  -webkit-transition-delay: .1s;
  transition-delay: .1s;
}

.c3 {
  display: inline-block;
  width: 1.125rem;
  height: 1.125rem;
  border: 1px solid #e1e5eb;
  border-radius: 2px;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  box-shadow: none;
  background-color: white;
  text-align: center;
}

input:checked + .c3 {
  background-color: #5a6169;
}

input:checked + .c3 svg {
  stroke: white;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div
  class="c0"
  disabled=""
  label=""
>
  <label
    class="c1"
    disabled=""
    label=""
  >
    <input
      class="c2"
      disabled=""
      label=""
      type="checkbox"
    />
    <div
      class="c3"
      disabled=""
      label=""
    >
      <svg
        class="c4"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
    <span
      class="c5"
      disabled=""
      label=""
    />
  </label>
</div>
`;
